crates/rspack_plugin_javascript/src/visitors/mod.rs

```diff
@@ -140,9 +150,10 @@ pub fn run_before_pass(
       // enable if configurable
       // swc_visitor::json_parse(min_cost),
       swc_visitor::paren_remover(comments.map(|v| v as &dyn Comments)),
+      // es_version
       swc_visitor::compat(
         options.builtins.preset_env.clone(),
-        None,
+        es_version,
```

Comment:
before https://github.com/web-infra-dev/rspack/blob/main/crates/rspack_plugin_javascript/src/visitors/swc_visitor/compat.rs#LL53C41-L53C41, ` compat_by_es_version` never run, because `es_version` is always `None`